crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -326,6 +449,18 @@ impl AlgorithmUpdaterV1 {
         );
     }
 
+    fn da_change_accounting_for_activity(&self, maybe_da_change: i128) -> i128 {
+        if maybe_da_change > 0 {
+            match self.l2_activity.safety_mode() {
+                DAGasPriceSafetyMode::Normal => maybe_da_change,
+                DAGasPriceSafetyMode::Hold => 0,
+                DAGasPriceSafetyMode::Decrease => self.max_change().saturating_mul(-1),
```

Comment:
Is it possible that `self.max_change() -> i128` gives a negative value and in such case we'll actually increase the price?